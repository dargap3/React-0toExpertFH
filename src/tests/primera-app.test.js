
import { shallow } from 'enzyme';
import PrimeraApp from "../primera-app";


describe('Pruebas en <PrimeraApp />', () => {

  /* test('should show the message, "hola hola"', () => {
    const saludo = 'hola hola';

    const { getByText } = render( <PrimeraApp saludo={ saludo } />);

    expect(getByText(saludo)).toBeInTheDocument();
  }); */

  test('should show <PrimeraApp /> correctly', () => {
    const saludo = "hola hola";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
    
  });

  test('should show subtitle sent it by props', () => {
    const saludo = "hola hola";
    const subtitulo = "soy carola"

    const wrapper = shallow(
        <PrimeraApp 
            saludo={saludo} 
            subtitulo={subtitulo}
        />
    );

    const textoParrafo = wrapper.find('p').text();

    expect(textoParrafo).toBe(subtitulo);  
  })
  
  
})
