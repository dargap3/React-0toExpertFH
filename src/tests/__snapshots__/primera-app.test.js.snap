// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    hola hola
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
